Sources/Store/ScopedStore.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  ScopedStore.swift
+//  ViewStore
+//
+//  Created by Kenneth Ackerson on 4/3/23.
+//
+
+import Foundation
+import Combine
+import CasePaths
+
+/// A `Store` that's purpose is to allow clients of it to modify a parent's Store and one of it's sub-stores without having direct access to either store.
+public final class ScopedStore<State, Action>: Store {
+    @Published public var state: State
+    
+    public var publishedState: AnyPublisher<State, Never> {
+        return $state.eraseToAnyPublisher()
+    }
+
+    private let action: (Action) -> Void
+    
+    /// Initializes a new `ScopedStore`
+    /// - Parameters:
+    ///   - initial: The initial state for this `Store`, likely a copy of whatever the current sub-store's state is now (see the `scoped` function on the `Store` extension for an example)
+    ///   - statePub: The publisher that allows this `ScopedStore` to get the lastest copy of the sub-store's state.
+    ///   - action: A closure to let you pass actions back to a parent `Store`. (see the `scoped` function on the `Store` extension for an example of embedding these into a "sub-action" of a parent Store to forward to a sub-store)
```

Comment:
Minor punctuation pass
```suggestion
    /// Initializes a new `ScopedStore`.
    /// - Parameters:
    ///   - initial: The initial state for this `Store`, likely a copy of whatever the current sub-store's state is now (see the `scoped` function on the `Store` extension for an example).
    ///   - statePub: The publisher that allows this `ScopedStore` to get the lastest copy of the sub-store's state.
    ///   - action: A closure to let you pass actions back to a parent `Store`. (see the `scoped` function on the `Store` extension for an example of embedding these into a "sub-action" of a parent Store to forward to a sub-store).
```